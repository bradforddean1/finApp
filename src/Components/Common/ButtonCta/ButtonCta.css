.ButtonCta {
	padding: 0;
	border-width: 0;
	border-radius: 2px;

	font-family: Sofia Pro;
	font-style: normal;
	font-weight: 500;
	font-size: 18px;
	line-height: 18px;
	color: #e5e5e5;
}

/* CTA */
.cta {
	background: #06497e;
	height: 50px;
}

a.cta {
	display: flex;
	justify-content: center;
	align-items: center;
}

/* Intermodal */
