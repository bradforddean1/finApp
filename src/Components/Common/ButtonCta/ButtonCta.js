import { createElement } from "react";
import "./ButtonCta.css";
import PropTypes from "prop-types";

/**
 * Big call to action button
 * @component
 */
function ButtonCta(props) {
	const { tag, children, className, ...otherProps } = props;

	return createElement(
		tag,
		{ className: ["ButtonCta", className].join(" "), ...otherProps },
		children
	);
}

ButtonCta.propTypes = {
	tag: PropTypes.string,
	children: PropTypes.string,
	className: PropTypes.string,
};

ButtonCta.defaultProps = {
	tag: "a",
};

export default ButtonCta;
