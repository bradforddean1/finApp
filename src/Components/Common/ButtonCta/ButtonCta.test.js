import React from "react";
import ReactDOM from "react-dom";
import ButtonCta from "./ButtonCta";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("ButtonCta Component", () => {
	it("Renders", () => {
		const div = document.createElement("div");
		ReactDOM.render(<ButtonCta />, div);
		ReactDOM.unmountComponentAtNode(div);
	});

	it("Renders the Default State", () => {
		const wrapper = shallow(<ButtonCta />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	const props = { tag: "a", children: "Test", className: "test" };

	it("Renders given props", () => {
		const wrapper = shallow(<ButtonCta {...props} />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
