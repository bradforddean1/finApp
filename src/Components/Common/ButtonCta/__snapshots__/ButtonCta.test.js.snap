// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonCta Component Renders given props 1`] = `
<a
  className="ButtonCta test"
>
  Test
</a>
`;

exports[`ButtonCta Component Renders the Default State 1`] = `
<a
  className="ButtonCta "
/>
`;
